docs/rules-from-web-poet.rst

```diff
@@ -187,21 +203,124 @@ Take note that since ``SCRAPY_POET_OVERRIDES`` is structured as
     section.
 
 
-Overrides registry
-==================
-
-The overrides registry is responsible for informing whether there exists an
-override for a particular type for a given request. The default overrides
-registry allows to configure these rules using patterns that follow the
-`url-matcher <https://url-matcher.readthedocs.io/en/stable/>`_ syntax. These rules can be configured using the
-``SCRAPY_POET_OVERRIDES`` setting, as it has been seen in the :ref:`intro-tutorial`
-example.
-
-But the registry implementation can be changed at convenience. A different
-registry implementation can be configured using the property
-``SCRAPY_POET_OVERRIDES_REGISTRY`` in ``settings.py``. The new registry
-must be a subclass of :class:`scrapy_poet.overrides.OverridesRegistryBase` and
-must implement the method :meth:`scrapy_poet.overrides.OverridesRegistryBase.overrides_for`.
-As other Scrapy components, it can be initialized from the ``from_crawler`` class
-method if implemented. This might be handy to be able to access settings, stats,
-request meta, etc.
+Item Returns
+============
+
+scrapy-poet also supports a convenient way of asking for items directly. This
+is made possible by the ``to_return`` parameter of :class:`web_poet.rules.ApplyRule`.
+The ``to_return`` specifies which item a page object is capable of returning for
+a given URL.
+
+Let's check out an example:
+
+.. code-block:: python
+
+    import attrs
+    import scrapy
+    from web_poet import WebPage, handle_urls, field
+
+
+    @attrs.define
+    class Image:
+        url: str
+
+
+    @handle_urls("example.com")
+    class ProductImagePage(WebPage[Image]):
+        @field
+        def url(self) -> str:
+            return self.css("#product img ::attr(href)").get("")
+
+
+    @attrs.define
+    class Product:
+        name: str
+        image: Image
+
+
+    @handle_urls("example.com")
+    @attrs.define
+    class ProductPage(WebPage[Product]):
+        # The ``Image`` class is declared as a dependency.
+        image: Image
+
+        @field
+        def name(self) -> str:
+            return self.css("h1.name ::text").get("")
+
+        @field
+        def image(self) -> Image:
+            return self.image
+
+
+    class MySpider(scrapy.Spider):
+        name = "myspider"
+        start_urls = ["https://example.com/products/some-product"]
+
+        # We can directly ask for the item here instead of the page object.
+        def parse(self, response, item: Product):
+            return item
+
+From this example, we can see that:
+
+    * Page objects can directly ask for items as dependencies.
+
+      This is made possible by the ``ApplyRule("example.com", use=ProductImagePage,
+      to_return=Image)`` instance created from the ``@handle_urls`` decorator
+      on ``ProductImagePage``.
+
+      The ``Image`` item instance directly comes from ``ProductImagePage`` (by
+      calling its ``.to_item()`` method behind the scenes) and is provided to
+      ``ProductPage``.
```

Comment:
I think it may make sense to drop this point altogether, and simplify the example above accordingly. Page object definition is the scope of web-poet, in scrapy-poet we should probably focus only on what is specific to scrapy-poet.

Similarly, we may be able to simplify this documentation by linking to the upstream web-poet docs for documentation on how to declare the item that a page object returns, and here focus only on how to use those in a Scrapy spider, i.e. simply cover that you can use the returned item as a type hint to get it.

Reply:
I think it'd be great to still frame both web-poet and scrapy-poet together in the same example. In that way, the user could get the whole picture (if they're new to this) about how both packages interact with one another.

Reply:
Makes sense.

But can we at least remove the code and explanation about page objects supporting items as dependencies? While I see how it is related, I don’t think it is relevant for this section about how Scrapy callbacks can ask for items. And removing it would simplify the section significantly.

Reply:
I see your point. Updated it in https://github.com/scrapinghub/scrapy-poet/pull/103/commits/dae69d8ba3e672732cc2105dbca77e8253b33ffe